src/ctx.rs

```diff
@@ -513,6 +516,72 @@ impl TryIntoCtx<Endian> for usize where usize: IntoCtx<Endian> {
     }
 }
 
+#[cfg(feature = "std")]
+impl<'a> TryFromCtx<'a> for &'a CStr {
+    type Error = error::Error;
+    type Size = usize;
+    #[inline]
+    fn try_from_ctx(src: &'a [u8], _ctx: ()) -> result::Result<(Self, Self::Size), Self::Error> {
+        let null_byte = match src.iter().position(|b| *b == 0) {
+            Some(ix) => ix,
+            None => return Err(error::Error::BadInput {
+                size: 0,
+                msg: "The input doesn't contain a null byte",
+            })
+        };
+
+        let data = src.get(..null_byte+1).unwrap();
```

Comment:
Why not just `src[..null_byte+1]`?

Also so I’m too stupid right now to reason about the +1 indexes and since there are tests checking bytes read I’ll just assume it’s correct :P

Reply:
I remember hitting a really bizarre lifetimes issue, where `&src[..null_byte + 1]` would reborrow from `src` giving `data` a temporary lifetime which only lasted until the end of the function. The `get()` method returned an `&[u8]` with the `'a` lifetime we want to return.

I believe I was writing this PR using `nightly` and when I tried to reproduce it in the playground everything worked :disappointed: Maybe there was a bug in NLL that's been fixed recently?

Also, the `+1` is because `..` is a half open range. When constructing the `&CStr` via `CStr::from_bytes_with_nul_unchecked(data)` we want to include that trailing `null`, so you need to specify the index *after* our null terminator.